Add README and bug fixes (#628)
rancher/charts: packages/backup-restore-operator/charts/README.md

# Backup-Restore-Operator Chart

> The backup-restore-operator provides ability to back up and restore Kubernetes cluster metadata for user applications 

Author: 👍 ! I'm using the second part of the suggestion, but I feel like `provides ability to back up` makes more sense

### Components

This chart installs the following clusterscoped CRDs  

* **Backup**  
  Creating an instance of the Backup CRD lets you perform a backup. A Backup CR must specify a ResourceSet, that contains a set of selectors to target certain resources/CRDs. (docs link TBD)
* **Restore**  
  Creating an instance of the Restore CRD lets you restore from a backup file. (docs link TBD)
* **ResourceSet**  

>   ResourceSet CR should contain a set of selectors for all Kubernetes resources and CRDs that your application installs and manages, so that the operator can gather them during backups (docs link TBD)

Author: makes sense, if I do `helm show readme` for this chart, markdown headers will look better

> ### User flow

Reviewer: Also, I feel like this section in general doesn't match my perception of a User Flow; imo this section should represent step by step what a user should do to use this chart, i.e.

- First configure where you want to store your backups
- Next define a ResourceSet to select resources to be installed
- Next define a Backup CR that refers to the prior created ResourceSet so that the operator can collect and save resources. Here you can specify choosing to encrypt resources before saving them
- Once you have your operator set up to collect backups, you can trigger a restore by creating the CR that specifies the specific backup you would like to restore your cluster to

> ### User flow

Reviewer: Maybe replace this with a configuration section like in `rancher-pushprox` [here](https://github.com/rancher/charts/blob/dev-v2.5/packages/rancher-pushprox/charts/README.md#configuration)

> 1. This chart can be configured to save all the backups at a particular location, such as s3/minio bucket, or in a Kubernetes Persistent Volume.

Reviewer: ```suggestion
1. This chart can be configured to save all the backups at a particular location. Currently, we support Amazon S3 / MinIO buckets or Kubernetes Persistent Volumes.
```

2. The Backup CR must refer a ResouceSet, and the operator will collect and save resources that match its ResourceSet
3. [This](linkTBD) is the ResourceSet that selects all resources installed by Rancher in the management cluster
The chart will add this ResourceSet in the cluster during installation
4. User can choose to encrypt resources before saving them in the backup (docs link TBD)
5. For restore, user must specify the exact tarball filename that contains the backup data.

### Installation
```

> helm install backup-restore-operator ./ -n cattle-resources-system --create-namespace 

Reviewer: shouldn't this be `helm install --create-namespace  -n cattle-resources-system backup-restore-operator rancher/stable`? Also, we should mention something about installing the CRD chart first here

Author: I'll add in rancher/stable. But there's no separate CRD chart, if user is using helm cli, the /crds folder will install them right?

Reviewer: Hmm maybe we should split the Installation into two sections:

*Note: added a backslash before the code block to allow Github to render a code block in a code block*

```
#### Using `rancher/stable`

If installing this from the `rancher/stable` repo, use the following commands
\```bash # 
helm install --create-namespace -n cattle-resources-system backup-restore-operator-crd rancher/stable
helm install --create-namespace -n cattle-resources-system backup-restore-operator rancher/stable
\```bash

The CRDs used by this chart are packaged in a separate chart in the `rancher/stable` repo to allow users to keep CRDs installed onto a cluster even if the main package is deleted. You can create, update, or delete the CRDs packaged in that chart by using the appropriate `helm install`, `helm upgrade`, and `helm uninstall` commands on the CRD chart.

#### Installing this chart from this repo

If downloading this chart from this repo directly, use
\```bash
helm install --create-namespace -n cattle-resources-system backup-restore-operator <path-to-chart> 
\```.

This form of the chart will package the CRDs directly within the chart. Since this uses Helm 3's `crd/` directory, these CRDs will only be installed onto the cluster and not upgraded / removed on a `helm upgrade` or `helm uninstall`. To clean up the CRDs from the chart on uninstall, please manually execute the following commands:

\```bash
kubectl -n cattle-resources-system delete crd <crd-name>
... (Note: repeat three times for each of the CRDs installed by this chart)
\```
```
